workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/internal/RealRenderContext.kt

```diff
@@ -51,8 +51,12 @@ internal class RealRenderContext<PropsT, StateT, OutputT>(
    * Used to:
    *  - prevent modifications to this object after [freeze] is called.
    *  - prevent sending to sinks before render returns.
+   *
+   * This is a [ThreadLocal] since we only care about preventing calls during rendering from the
```

Comment:
And we're confident that we only ever call `freeze()` from the render thread? I can't think of any violations.

Reply:
Ya I can't think of `RenderContext` methods that would be called off main thread? will keep thinking.